src/common/config/init.js

```diff
@@ -3,15 +3,95 @@
+ * @property {boolean} [ajax.enabled] - Turn on/off the ajax feature (on by default).
+ * @property {boolean} [ajax.autoStart] - If true, the agent will automatically start the ajax feature. Otherwise, it will be in a deferred state until the `start` API method is called.
+ * @property {Object} [distributed_tracing]
+ * @property {boolean} [distributed_tracing.enabled] - If true, distributed tracing headers will be added to outgoing requests. Requires ajax feature to be running.
+ * @property {boolean} [distributed_tracing.exclude_newrelic_header]
+ * @property {boolean} [distributed_tracing.cors_use_newrelic_header]
+ * @property {boolean} [distributed_tracing.cors_use_tracecontext_headers]
+ * @property {Array<string>} [distributed_tracing.allowed_origins]
+ * @property {Array<string>} [feature_flags] - An array of feature flags to enable experimental features.
+ * @property {Object} [generic_events]
+ * @property {boolean} [generic_events.enabled] - Turn on/off the generic events feature (on by default). This is required for `PageAction`, `UserAction`, and `BrowserPerformance` events.
+ * @property {boolean} [generic_events.autoStart] - If true, the agent will automatically start the generic events feature. Otherwise, it will be in a deferred state until the `start` API method is called.
+ * @property {Object} [harvest]
+ * @property {number} [harvest.interval] - The interval in seconds at which the agent will send out data. It's not recommended to change this value.
+ * @property {Object} [jserrors]
+ * @property {boolean} [jserrors.enabled] - Turn on/off the jserrors feature (on by default).
+ * @property {boolean} [jserrors.autoStart] - If true, the agent will automatically start the jserrors feature. Otherwise, it will be in a deferred state until the `start` API method is called.
+ * @property {Object} [logging]
+ * @property {boolean} [logging.enabled] - Turn on/off the logging feature (on by default).
+ * @property {boolean} [logging.autoStart] - If true, the agent will automatically start the logging feature. Otherwise, it will be in a deferred state until the `start` API method is called.
+ * @property {Object} [metrics]
+ * @property {boolean} [metrics.enabled] - Turn on/off the metrics feature (on by default).
+ * @property {boolean} [metrics.autoStart] - If true, the agent will automatically start the metrics feature. Otherwise, it will be in a deferred state until the `start` API method is called.
+ * @property {Array<Object>} [obfuscate] - Array of regexp and corresponding replacement patterns for obfuscating data.
+ * @property {Object} [page_action]
+ * @property {boolean} [page_action.enabled] - Must be true to allow PageAction events to be captured.
+ * @property {Object} [page_view_event]
+ * @property {boolean} [page_view_event.enabled] - This setting is ignored! PageViewEvent is always enabled by force.
```

Comment:
we should discuss if these types of configs (not intended for public mutation) are worth defining

Reply:
I think if it isn't used at all, it should be removed to begin with. The typings don't actually prevent people from plugging in the property anyways in JS. It's really just informational.